airbyte-ci/connectors/pipelines/pipelines/connector_changes/base_image_version_migration.py

```diff
@@ -0,0 +1,322 @@
+            When contributing on our connector you might need to customize the build process to add a system dependency or set an env var.
+            You can customize our build process by adding a `build_customization.py` module to your connector.
+            This module should contain a `pre_connector_install` and `post_connector_install` async function that will mutate the base image and the connector container respectively.
+            It will be imported at runtime by our build process and the functions will be called if they exist.
+
+            Here is an example of a `build_customization.py` module:
+            ```python
+            from __future__ import annotations
+
+            from typing import TYPE_CHECKING
+
+            if TYPE_CHECKING:
+                # Feel free to check the dagger documentation for more information on the Container object and its methods.
+                # https://dagger-io.readthedocs.io/en/sdk-python-v0.6.4/
+                from dagger import Container
+
+
+            async def pre_connector_install(base_image_container: Container) -> Container:
+                return await base_image_container.with_env_variable("MY_PRE_BUILD_ENV_VAR", "my_pre_build_env_var_value")
+
+            async def post_connector_install(connector_container: Container) -> Container:
+                return await connector_container.with_env_variable("MY_POST_BUILD_ENV_VAR", "my_post_build_env_var_value")
+            ```
+            """
+            )
+        )
+
+        build_instructions = build_instructions_template.render({"connector_image": self.context.connector.metadata["dockerRepository"]})
+
+        new_doc = "\n".join(og_lines[:line_no_for_build_instructions] + [build_instructions] + og_lines[line_no_for_build_instructions:])
+        return new_doc
+
+
+async def run_connector_base_image_upgrade_pipeline(context: ConnectorContext, semaphore, set_if_not_exists: bool) -> ConnectorReport:
+    """Run a pipeline to upgrade for a single connector to use our base image."""
+    async with semaphore:
+        steps_results = []
+        async with context:
+            og_repo_dir = await context.get_repo_dir()
+            update_base_image_in_metadata = UpgradeBaseImageMetadata(
+                context,
+                og_repo_dir,
+                set_if_not_exists=set_if_not_exists,
+            )
+            update_base_image_in_metadata_result = await update_base_image_in_metadata.run()
+            steps_results.append(update_base_image_in_metadata_result)
+            final_repo_dir = update_base_image_in_metadata_result.output_artifact
+            await og_repo_dir.diff(final_repo_dir).export(str(consts.REPO_PATH))
+            context.report = ConnectorReport(context, steps_results, name="BASE IMAGE UPGRADE RESULTS")
+    return context.report
+
+
+async def run_connector_migration_to_base_image_pipeline(context: ConnectorContext, semaphore, pull_request_number: str):
```

Comment:
❓Why a pipeline and not a script?

Fine for it to stay as a pipeline but we likely want to add a note in a comment for when we can delete this code.

Reply:
I'm not sure this is a short lived command, as long as community connector use Dockerfile it has a reason to be considered a pipeline and not adhoc script.